src/noise/session.rs

```diff
@@ -128,10 +136,22 @@ impl ReceivingKeyCounterValidator {
             return Ok(());
         }
         // Packets where dropped, or maybe reordered, skip them and mark unused
-        let mut i = self.next;
-        while i < counter {
-            self.clear_bit(i);
-            i += 1;
+        if counter - self.next >= N_BITS {
+            // Too far ahead, clear all the bits
+            for c in self.bitmap.iter_mut() {
+                *c = 0;
+            }
+        } else {
+            let mut i = self.next;
+            while i + WORD_SIZE < counter {
+                // Clear whole word at a time
+                self.clear_word(i);
```

Comment:
Won't this also clear bits before and including "next" if "next" does not align on a word boundary?

I think that another loop is needed before:
```rust
while i % WORD_SIZE != 0 && i < counter {
    self.clear_bit(i);
    i += 1;
}
```

(+need test case)

Reply:
Fixed now